pkg/fqdn/namemanager/manager.go

```diff
@@ -467,3 +487,70 @@ func (n *manager) mapSelectorsToNamesLocked(fqdnSelector api.FQDNSelector) (name
 func prepareMatchName(matchName string) string {
 	return dns.FQDN(matchName)
 }
+
+func (n *manager) triggerAllocateIdentity(sel api.FQDNSelector) {
+	n.allocLock.Lock()
+	defer n.allocLock.Unlock()
+	n.allocSelectors = append(n.allocSelectors, sel)
+	n.allocTrigger.Trigger()
+}
+
+func (n *manager) allocateIdentity(_ []string) {
+	// The assumption is that this context is never used, since these are
+	// locally-scoped identities.
+	ctx := context.Background()
+	n.allocLock.Lock()
+	defer n.allocLock.Unlock()
+	for _, sel := range n.allocSelectors {
+		lbls := identityLabels(sel)
+		for _, ls := range lbls {
+			_, _, err := n.params.IdentityAllocator.AllocateIdentity(ctx, ls, true, identity.InvalidIdentity)
```

Comment:
You can use `AllocateLocalIdentity()` which doesn't require a context. It just needs to be added to the `IdentityAllocator` interface.

Reply:
`AllocateLocalIdentity` is not part of the `IdentityAllocator` interface. It looks like  `AllocateIdentity` immediately checks the labels. `AllocateIdentities` immediately calls `needsGlobalIdentity` to see if the labels can be locally allocated. I would prefer to use that logic than introduce a new method into the `IdentityAllocator` interface.